src/types/package.go

```diff
@@ -21,7 +21,7 @@ type ZarfPackage struct {
 	Kind       ZarfPackageKind                 `json:"kind" jsonschema:"description=The kind of Zarf package,enum=ZarfInitConfig,enum=ZarfPackageConfig,default=ZarfPackageConfig"`
 	Metadata   ZarfMetadata                    `json:"metadata,omitempty" jsonschema:"description=Package metadata"`
 	Build      ZarfBuildData                   `json:"build,omitempty" jsonschema:"description=Zarf-generated package build data"`
-	Components []ZarfComponent                 `json:"components" jsonschema:"description=List of components to deploy in this package"`
+	Components []ZarfComponent                 `json:"components" jsonschema:"description=List of components to deploy in this package,minItems=1"`
```

Comment:
Yeah it's very nice. There's more we can add to the schema as well, but wanted to keep it simple in this PR. I think we can also do it in code rather than in json comments which will probably be easier to read for anything complex.